### PR TITLE
SponsorBlock Integration

### LINKED ISSUE #18: SponsorBlock Support
This is a nice to have, Piped's API supports proxying this (the K-Anonymity model is followed when requesting data from sponsor.ajay.app).
For example: https://pipedapi.kavin.rocks/sponsors/EE4OCYOLjBc?category=%5B%22sponsor%22%2C%22interaction%22%2C%22selfpromo%22%2C%22music_offtopic%22%5D

### PR DESCRIPTION
**Pull request by janisslsm**
Integrates sponsorblock. Currently segments aren't show on the seek bar cause it requires heavy player modifications which isn't the purpose of this PR.
Closes #18

### CODE REVIEW

**Comment by FireMasterK** [comment]
You should add a link to SponsorBlock somewhere: https://github.com/ajayyy/SponsorBlock/wiki/Database-and-API-License
In Piped, it was added like so: https://github.com/TeamPiped/Piped/pull/90

File: app/src/main/java/com/github/libretube/MainActivity.kt [hunk 1]
```diff
@@ -44,6 +44,13 @@ class MainActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this)
         RetrofitInstance.url = sharedPreferences.getString("instance", "https://pipedapi.kavin.rocks/")!!
+        SponsorBlockSettings.sponsorBlockEnabled = sharedPreferences.getBoolean("sponsorblock_enabled_key", false)
+        SponsorBlockSettings.introEnabled = sharedPreferences.getBoolean("intro_category_key", false)
+        SponsorBlockSettings.selfPromoEnabled = sharedPreferences.getBoolean("selfpromo_category_key", false)
+        SponsorBlockSettings.interactionEnabled = sharedPreferences.getBoolean("interaction_category_key", false)
+        SponsorBlockSettings.sponsorsEnabled = sharedPreferences.getBoolean("sponsors_category_key", false)
+        SponsorBlockSettings.outroEnabled = sharedPreferences.getBoolean("outro_category_key", false)
```
Comment (FireMasterK) [inline app/src/main/java/com/github/libretube/MainActivity.kt:52]: @lilcheti do we want to enable some by default? In Piped, I selected a few based on what most people might want by default

**Review by FireMasterK — APPROVED** [review]
From a code review perspective, LGTM!

**Comment by janisslsm** [comment]
> You should add a link to SponsorBlock somewhere: https://github.com/ajayyy/SponsorBlock/wiki/Database-and-API-License In Piped, it was added like so: [TeamPiped/Piped#90](https://github.com/TeamPiped/Piped/pull/90)

@FireMasterK This should be fine?
image

**Comment by FireMasterK** [comment]
> @FireMasterK This should be fine?
> image

Yes, that's perfect!

**Comment by IceDBorn** [comment]
It works great!

**Pull request merged**